### PR TITLE
Tables PR

### PR DESCRIPTION
**Pull request by AngelAvilaGit**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/tables/tables.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+# **Tables**
+
+This library is internal, and will automatically work when referencing the **global** keyword **table**. In Lua, **Tables** refer to **Arrays**. <!-- Arrays HREF -->
+
+## **Functions**
+
+Below are the list of Table methods available.
+
+### table.clear(table : **table**) : void
```
Comment (lucwhite) [inline content/lua/concepts/tables/tables.md:24]: Make note in the handoff doc that these should likely be terms in the future.

File: content/lua/concepts/tables/tables.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Computer Science'
+  - 'Lua'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Tables'
+  - 'Arrays'
+  - 'Libraries'
+  - 'Methods'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'lua/tables'
```
Comment (lucwhite) [inline content/lua/concepts/tables/tables.md:13]:
```suggestion
  - 'learn-lua'
```

This should always be `learn-lua` as that's what the corresponding course name is.

File: content/lua/concepts/tables/tables.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+---
+
+# **Tables**
+
+This library is internal, and will automatically work when referencing the **global** keyword **table**. In Lua, **Tables** refer to **Arrays**. <!-- Arrays HREF -->
```
Comment (lucwhite) [inline content/lua/concepts/tables/tables.md:18]: Issue: According to the [Lua Documentation](https://www.lua.org/pil/2.5.html) I don't think its perfectly accurate to say that tables refer to arrays, as they're similar but don't follow all the same rules. Consider rephrasing this a bit. Reach out on slack if you need additional guidance.

Also, the link would likely be `[Arrays](https://www.codecademy.com/resources/docs/lua/arrays)`

File: content/lua/concepts/tables/tables.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## **Functions**
+
+Below are the list of Table methods available.
+
+### table.clear(table : **table**) : void
+Clears all Indexes and Values from a table.
+
+### table.clone(table : **table**) : table
+Returns a shallow copy of the provided table
```
Comment (lucwhite) [inline content/lua/concepts/tables/tables.md:28]:
```suggestion
Returns a shallow copy of the provided table.
```

Missing an ending period. You might also want to elaborate a bit on what a shallow copy means here.

File: content/lua/concepts/tables/tables.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+Returns a new table with the provided value repeated the provided amount of times.
+
+### table.find(haystack: **table**, needle : any, i : number?) : number
+Returns the requested needle’s index if found or returns `nil` if the needle does not exist within the provided haystack table.
+
+### table.foreach(table : **table**, callback : function) : void
+Iterates through the provided table using the provided callback function. The provided function will receive two parameters, of which are **Index** and **Value** from the current table iteration.
+
+### table.freeze(table : **table**) : void
+Freezes the provided table, and makes it **`read-only`**
```
Comment (lucwhite) [inline content/lua/concepts/tables/tables.md:43]:
```suggestion
Freezes the provided table, and makes it **`read-only`**.
```

**Review by lucwhite — CHANGES_REQUESTED** [review]
Great work Angel, left a few suggestions, let me know if you have any questions!

**Comment by SSwiniarski** [comment]
@AngelAvilaGit please remember to sign the [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=2926)

File: content/lua/concepts/tables/tables.md (hunk 3 above)
Comment (AngelAvilaGit) [reply content/lua/concepts/tables/tables.md:18]: I reviewed the Lua documentation and rephrased it to fit the behavior of a Table more adequately. Thanks!

**Comment by AngelAvilaGit** [comment]
> @AngelAvilaGit please remember to sign the [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=2926)

Signed!

File: content/lua/concepts/tables/tables.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
+---
+Title: 'Tables' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
+Description: 'Tables are used to store collections of data.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Computer Science'
+  - 'Lua'
```
Comment (lucwhite) [inline content/lua/concepts/tables/tables.md:6]:
```suggestion
```

Lua actually is not a subject

File: content/lua/concepts/tables/tables.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Tables'
+  - 'Arrays'
+  - 'Libraries'
+  - 'Methods'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-lua'
```
Comment (lucwhite) [inline content/lua/concepts/tables/tables.md:13]:
```suggestion
  - 'learn-lua'
  - 'paths/computer-science'
```

Paul suggested including this path as well.

**Review by lucwhite — APPROVED** [review]
Looks good to me! Ready for 2nd review

**Comment by AngelAvilaGit** [comment]
> @AngelAvilaGit @lucwhite From what I've read, Lua doesn't really have arrays, only tables. I think the arrays entry should be renamed tables, and a terms folder can be added in the future that contains the methods documented here.

In Lua, we generally call Tables Arrays as `table` itself is a global term. When I mention arrays, I'm referring to numerically indexed arrays as opposed to associative arrays which follow a `Key:Value` format instead of a `number:Value` format.

Should I rename/revise anything on my end?

**Comment by lucwhite** [comment]
> In Lua, we generally call Tables Arrays as `table` itself is a global term. When I mention arrays, I'm referring to numerically indexed arrays as opposed to associative arrays which follow a `Key:Value` format instead of a `number:Value` format.
> 
> Should I rename/revise anything on my end?

[For context](https://www.lua.org/pil/11.1.html)

**Review by yangc95 — APPROVED** [review]
I made some direct changes @AngelAvilaGit LGTM! 

Also, I used your exact wordings in creating the follow-up issues for the functions that will populate here as table function terms.

**Pull request merged**